Library/Homebrew/brew.sh

```diff
@@ -516,13 +516,13 @@ esac
 # - https://github.com/Homebrew/install/blob/HEAD/install.sh
 # and, if needed:
 # - MacOSVersion::SYMBOLS
-HOMEBREW_MACOS_NEWEST_UNSUPPORTED="15"
+HOMEBREW_MACOS_NEWEST_UNSUPPORTED="16"
 # TODO: bump version when new macOS is released and update references in:
 # - docs/Installation.md
 # - HOMEBREW_MACOS_OLDEST_SUPPORTED in .github/workflows/pkg-installer.yml
 # - `os-version min` in package/Distribution.xml
 # - https://github.com/Homebrew/install/blob/HEAD/install.sh
-HOMEBREW_MACOS_OLDEST_SUPPORTED="12"
+HOMEBREW_MACOS_OLDEST_SUPPORTED="13"
```

Comment:
FYI, this change will stop all Monterey CI in Homebrew/core.

That will happen eventually, of course. Just mentioning this since it's relevant to the timing of merging this.

Reply:
Doing it separately makes sense to me.

Reply:
We should stop Monterey CI as soon as we start any Sequoia CI.

Reply:
Can you elaborate a bit more on what "separate these controlling CI " and "separating CI control" means here? It's not totally clear. Thanks!

Reply:
These variables currently not only control warnings etc but also directly control which macOS versions are used in homebrew-core CI. So if we drop macOS 12 support here then it immediately drops macOS 12 CI from Homebrew/core even though we haven't tagged yet.

Reply:
Thanks for explaining. Sounds like the main blocker therefore is updating the CI runners to not use the same logic as these numbers here.

Reply:
Unless someone else does, I'll find some time tomorrow to do this. I agree this should be unblocked and merged as soon as possible.

Reply:
I'll pick this up today.

Reply:
Thanks!

Reply:
#18306